From 5e6bb742f869a40c60739d58fd7cd9def09d3422 Mon Sep 17 00:00:00 2001
From: Nguyen Tran <nguyen.tran.pz@bp.renesas.com>
Date: Tue, 2 Dec 2025 14:31:21 +0700
Subject: [PATCH 3/3] arm64: dts: renesas: r8a779g{0,3,4,5}-native-rgid: Add
 Native device support

In the default RGID environment, all UIO devices were enabled, Native
devices (such as DU devices, GPU, ...) that was conflicted with
UIO devices.

The patch will create a new Device Tree to support enabling Native
devices that support for display output, GPU, and basic peripherals.

For convenience to use:
- "r8a779g{0, 3, 4, 5,}-whitehawk-rgid.dtb": All UIO devices enable, Native devices
  disabled.
- "r8a779g{0, 3, 4, 5}-whitehawk-native-rgid.dtb": All UIO devices enable except for
  display
output, GPU and basic peripherals.

Signed-off-by: Nguyen Tran <nguyen.tran.pz@bp.renesas.com>
---
 arch/arm64/boot/dts/renesas/Makefile          |   5 +
 .../r8a779g0-whitehawk-native-rgid.dts        |  19 +++
 .../r8a779g3-whitehawk-native-rgid.dts        |  20 +++
 .../r8a779g4-whitehawk-native-rgid.dts        |  20 +++
 .../r8a779g5-whitehawk-native-rgid.dts        |  20 +++
 .../dts/renesas/white-hawk-native-rgid.dtsi   | 107 ++++++++++++++++
 .../boot/dts/renesas/white-hawk-native.dtsi   | 114 ++++++++++++++----
 7 files changed, 282 insertions(+), 23 deletions(-)
 create mode 100644 arch/arm64/boot/dts/renesas/r8a779g0-whitehawk-native-rgid.dts
 create mode 100644 arch/arm64/boot/dts/renesas/r8a779g3-whitehawk-native-rgid.dts
 create mode 100644 arch/arm64/boot/dts/renesas/r8a779g4-whitehawk-native-rgid.dts
 create mode 100644 arch/arm64/boot/dts/renesas/r8a779g5-whitehawk-native-rgid.dts
 create mode 100644 arch/arm64/boot/dts/renesas/white-hawk-native-rgid.dtsi

diff --git a/arch/arm64/boot/dts/renesas/Makefile b/arch/arm64/boot/dts/renesas/Makefile
index ad3eb7bab3901..c2f5eeefa4d7f 100644
--- a/arch/arm64/boot/dts/renesas/Makefile
+++ b/arch/arm64/boot/dts/renesas/Makefile
@@ -79,6 +79,11 @@ dtb-$(CONFIG_ARCH_R8A779G0) += r8a779g3-whitehawk-native.dtb r8a779g3-whitehawk-
 dtb-$(CONFIG_ARCH_R8A779G0) += r8a779g4-whitehawk-native.dtb r8a779g4-whitehawk-du-native.dtb
 dtb-$(CONFIG_ARCH_R8A779G0) += r8a779g5-whitehawk-native.dtb r8a779g5-whitehawk-du-native.dtb
 
+dtb-$(CONFIG_ARCH_R8A779G0) += r8a779g0-whitehawk-native-rgid.dtb
+dtb-$(CONFIG_ARCH_R8A779G0) += r8a779g3-whitehawk-native-rgid.dtb
+dtb-$(CONFIG_ARCH_R8A779G0) += r8a779g4-whitehawk-native-rgid.dtb
+dtb-$(CONFIG_ARCH_R8A779G0) += r8a779g5-whitehawk-native-rgid.dtb
+
 dtb-$(CONFIG_ARCH_R8A779G0) += r8a779g3-white-hawk.dtb r8a779g3-white-hawk-rgid.dtb
 dtb-$(CONFIG_ARCH_R8A779G0) += r8a779g3-whitehawk-sound.dtb r8a779g3-whitehawk-sound-rgid.dtb
 
diff --git a/arch/arm64/boot/dts/renesas/r8a779g0-whitehawk-native-rgid.dts b/arch/arm64/boot/dts/renesas/r8a779g0-whitehawk-native-rgid.dts
new file mode 100644
index 0000000000000..46506fddf8833
--- /dev/null
+++ b/arch/arm64/boot/dts/renesas/r8a779g0-whitehawk-native-rgid.dts
@@ -0,0 +1,19 @@
+// SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+/*
+ * Device Tree Source for the R-Car V4H White Hawk board
+ * Support Native RGID driver.
+ *
+ * Copyright (C) 2025 Renesas Electronics Corp.
+ */
+
+/dts-v1/;
+#include "white-hawk-native-rgid.dtsi"
+#include "r8a779g0-rgid.dtsi"
+#include "white-hawk-cpu-common-rgid.dtsi"
+#include "white-hawk-common-rgid.dtsi"
+#include "white-hawk-native-common.dtsi"
+
+/ {
+	model = "Renesas White Hawk board based on r8a779g0";
+	compatible = "renesas,whitehawk", "renesas,r8a779g0";
+};
diff --git a/arch/arm64/boot/dts/renesas/r8a779g3-whitehawk-native-rgid.dts b/arch/arm64/boot/dts/renesas/r8a779g3-whitehawk-native-rgid.dts
new file mode 100644
index 0000000000000..d95f7e02c8d00
--- /dev/null
+++ b/arch/arm64/boot/dts/renesas/r8a779g3-whitehawk-native-rgid.dts
@@ -0,0 +1,20 @@
+// SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+/*
+ * Device Tree Source for the R-Car V4H-7 White Hawk board
+ * Support Native RGID driver.
+ *
+ * Copyright (C) 2025 Renesas Electronics Corp.
+ */
+
+/dts-v1/;
+#include "white-hawk-native-rgid.dtsi"
+#include "r8a779g3-rgid.dtsi"
+#include "white-hawk-cpu-common-rgid.dtsi"
+#include "white-hawk-common-rgid.dtsi"
+#include "white-hawk-native-common.dtsi"
+
+/ {
+	model = "Renesas White Hawk board based on r8a779g0";
+	compatible = "renesas,whitehawk", "renesas,r8a779g0",
+				"renesas,r8a779g3";
+};
diff --git a/arch/arm64/boot/dts/renesas/r8a779g4-whitehawk-native-rgid.dts b/arch/arm64/boot/dts/renesas/r8a779g4-whitehawk-native-rgid.dts
new file mode 100644
index 0000000000000..9307f2705689e
--- /dev/null
+++ b/arch/arm64/boot/dts/renesas/r8a779g4-whitehawk-native-rgid.dts
@@ -0,0 +1,20 @@
+// SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+/*
+ * Device Tree Source for the R-Car V4H-5 White Hawk board
+ * Support Native RGID driver.
+ *
+ * Copyright (C) 2025 Renesas Electronics Corp.
+ */
+
+/dts-v1/;
+#include "white-hawk-native-rgid.dtsi"
+#include "r8a779g4-rgid.dtsi"
+#include "white-hawk-cpu-common-rgid.dtsi"
+#include "white-hawk-common-rgid.dtsi"
+#include "white-hawk-native-common.dtsi"
+
+/ {
+	model = "Renesas White Hawk board based on r8a779g0";
+	compatible = "renesas,whitehawk", "renesas,r8a779g0",
+				"renesas,r8a779g4";
+};
diff --git a/arch/arm64/boot/dts/renesas/r8a779g5-whitehawk-native-rgid.dts b/arch/arm64/boot/dts/renesas/r8a779g5-whitehawk-native-rgid.dts
new file mode 100644
index 0000000000000..013db6fd6bf1b
--- /dev/null
+++ b/arch/arm64/boot/dts/renesas/r8a779g5-whitehawk-native-rgid.dts
@@ -0,0 +1,20 @@
+// SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+/*
+ * Device Tree Source for the R-Car V4H-3 White Hawk board
+ * Support Native RGID driver.
+ *
+ * Copyright (C) 2025 Renesas Electronics Corp.
+ */
+
+/dts-v1/;
+#include "white-hawk-native-rgid.dtsi"
+#include "r8a779g5-rgid.dtsi"
+#include "white-hawk-cpu-common-rgid.dtsi"
+#include "white-hawk-common-rgid.dtsi"
+#include "white-hawk-native-common.dtsi"
+
+/ {
+	model = "Renesas White Hawk board based on r8a779g0";
+	compatible = "renesas,whitehawk", "renesas,r8a779g0",
+				"renesas,r8a779g5";
+};
diff --git a/arch/arm64/boot/dts/renesas/white-hawk-native-rgid.dtsi b/arch/arm64/boot/dts/renesas/white-hawk-native-rgid.dtsi
new file mode 100644
index 0000000000000..c3e70d31cfb3f
--- /dev/null
+++ b/arch/arm64/boot/dts/renesas/white-hawk-native-rgid.dtsi
@@ -0,0 +1,107 @@
+// SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+/*
+ * Device Tree Source support Native RGID for the White Hawk CPU
+ * boards with R-Car V4H
+ *
+ * Copyright (C) 2025 Renesas Electronics Corp.
+ */
+
+#include "r8a779g0-rgid.dtsi"
+
+/ {
+	compatible = "renesas,r8a779g0-native", "renesas,r8a779g0";
+
+	soc: soc {
+		du0 {
+			status = "disabled";
+		};
+
+		du_01 {
+			status = "disabled";
+		};
+
+		dsi0 {
+			status = "disabled";
+		};
+
+		dsi0_01 {
+			status = "disabled";
+		};
+
+		dsi0_02 {
+			status = "disabled";
+		};
+
+		dsi0_03 {
+			status = "disabled";
+		};
+
+		dsi0_04 {
+			status = "disabled";
+		};
+
+		dsi0_05 {
+			status = "disabled";
+		};
+
+		dsi0_06 {
+			status = "disabled";
+		};
+
+		dsi0_07 {
+			status = "disabled";
+		};
+
+		dsi0_08 {
+			status = "disabled";
+		};
+
+		dsi1 {
+			status = "disabled";
+		};
+
+		dsi1_01 {
+			status = "disabled";
+		};
+
+		dsi1_02 {
+			status = "disabled";
+		};
+
+		dsi1_03 {
+			status = "disabled";
+		};
+
+		dsi1_04 {
+			status = "disabled";
+		};
+
+		dsi1_05 {
+			status = "disabled";
+		};
+
+		dsi1_06 {
+			status = "disabled";
+		};
+
+		dsi1_07 {
+			status = "disabled";
+		};
+
+		dsi1_08 {
+			status = "disabled";
+		};
+
+		doc0: doc@feba0000 {
+			status = "disabled";
+		};
+
+		doc0_01 {
+			status = "disabled";
+		};
+
+		dsc0: dsc@feb8d000 {
+			status = "disabled";
+		};
+	};
+};
diff --git a/arch/arm64/boot/dts/renesas/white-hawk-native.dtsi b/arch/arm64/boot/dts/renesas/white-hawk-native.dtsi
index 84ade98e4b343..d411545e38f69 100644
--- a/arch/arm64/boot/dts/renesas/white-hawk-native.dtsi
+++ b/arch/arm64/boot/dts/renesas/white-hawk-native.dtsi
@@ -12,28 +12,96 @@ / {
 	compatible = "renesas,r8a779g0-native", "renesas,r8a779g0";
 
 	soc: soc {
-		/delete-node/du0;
-		/delete-node/du_01;
-		/delete-node/dsi0;
-		/delete-node/dsi0_01;
-		/delete-node/dsi0_02;
-		/delete-node/dsi0_03;
-		/delete-node/dsi0_04;
-		/delete-node/dsi0_05;
-		/delete-node/dsi0_06;
-		/delete-node/dsi0_07;
-		/delete-node/dsi0_08;
-		/delete-node/dsi1;
-		/delete-node/dsi1_01;
-		/delete-node/dsi1_02;
-		/delete-node/dsi1_03;
-		/delete-node/dsi1_04;
-		/delete-node/dsi1_05;
-		/delete-node/dsi1_06;
-		/delete-node/dsi1_07;
-		/delete-node/dsi1_08;
-		/delete-node/doc@feba0000;
-		/delete-node/doc0_01;
-		/delete-node/dsc@feb8d000;
+		du0 {
+			status = "disabled";
+		};
+
+		du_01 {
+			status = "disabled";
+		};
+
+		dsi0 {
+			status = "disabled";
+		};
+
+		dsi0_01 {
+			status = "disabled";
+		};
+
+		dsi0_02 {
+			status = "disabled";
+		};
+
+		dsi0_03 {
+			status = "disabled";
+		};
+
+		dsi0_04 {
+			status = "disabled";
+		};
+
+		dsi0_05 {
+			status = "disabled";
+		};
+
+		dsi0_06 {
+			status = "disabled";
+		};
+
+		dsi0_07 {
+			status = "disabled";
+		};
+
+		dsi0_08 {
+			status = "disabled";
+		};
+
+		dsi1 {
+			status = "disabled";
+		};
+
+		dsi1_01 {
+			status = "disabled";
+		};
+
+		dsi1_02 {
+			status = "disabled";
+		};
+
+		dsi1_03 {
+			status = "disabled";
+		};
+
+		dsi1_04 {
+			status = "disabled";
+		};
+
+		dsi1_05 {
+			status = "disabled";
+		};
+
+		dsi1_06 {
+			status = "disabled";
+		};
+
+		dsi1_07 {
+			status = "disabled";
+		};
+
+		dsi1_08 {
+			status = "disabled";
+		};
+
+		doc0: doc@feba0000 {
+			status = "disabled";
+		};
+
+		doc0_01 {
+			status = "disabled";
+		};
+
+		dsc0: dsc@feb8d000 {
+			status = "disabled";
+		};
 	};
 };
\ No newline at end of file
-- 
2.25.1

